android/src/main/java/com/christopherdro/htmltopdf/RNHTMLtoPDFModule.java

```diff
@@ -32,57 +42,59 @@ public String getName() {
   public void convert(final ReadableMap options, final Promise promise) {
     try {
       File destinationFile;
-      String htmlString = options.hasKey("html") ? options.getString("html") : null;
-      if (htmlString == null) return;
+      String htmlString = options.hasKey(HTML) ? options.getString(HTML) : null;
+      if (htmlString == null) {
+        promise.reject(new Exception("RNHTMLtoPDF error: Invalid htmlString parameter."));
+        return;
+      }
 
       String fileName;
-      if (options.hasKey("fileName")) {
-        fileName = options.getString("fileName");
+      if (options.hasKey(FILE_NAME)) {
+        fileName = options.getString(FILE_NAME);
+        if (!isFileNameValid(fileName)) {
+          promise.reject(new Exception("RNHTMLtoPDF error: Invalid fileName parameter."));
+          return;
+        }
       } else {
         fileName = UUID.randomUUID().toString();
       }
 
-      if (options.hasKey("directory") && options.getString("directory").equals("docs")) {
+      if (options.hasKey(DIRECTORY)) {
         String state = Environment.getExternalStorageState();
-          File path = (Environment.MEDIA_MOUNTED.equals(state)) ?
-                  new File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_DOCUMENTS)
-                  : new File(mReactContext.getFilesDir(), Environment.DIRECTORY_DOCUMENTS);
+        File path = (Environment.MEDIA_MOUNTED.equals(state)) ?
+          new File(Environment.getExternalStorageDirectory(), options.getString(DIRECTORY)) :
+          new File(mReactContext.getFilesDir(), options.getString(DIRECTORY));
```

Comment:
What if the directory provided doesn't exist?

Maybe we expose props for the following fields.
https://developer.android.com/reference/android/os/Environment#fields_2

Reply:
The directory will be created, if it doesn't exist. 

That is one of the feature I was looking for. I need to create a file in a custom directory structure on sdcard (not the standard Android directories).